### ISSUE TITLE
[Bug]: QuickSight dataset state has constant changes to apply

### ISSUE THREAD

**Issue by g-dx** [hashicorp/terraform-provider-aws#30705]
### Terraform Core Version

1.4.5

### AWS Provider Version

4.62.0

### Affected Resource(s)

`aws_quicksight_data_set`

### Expected Behavior

Once applied and no changes have been made to the remote or local files there should be no changes in any subsequent plans.

### Actual Behavior

Any new plans always show changes even though nothing has changed.

Below is typical output:

```shell
aws_quicksight_data_set.test: Refreshing state... [id=ACCOUNT_ID,test]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_quicksight_data_set.test will be updated in-place
  ~ resource "aws_quicksight_data_set" "test" {
        id             = "ACCOUNT_ID,test"
        name           = "Test"
        tags           = {}
        # (5 unchanged attributes hidden)

      - logical_table_map {
          - alias                = "table" -> null
          - logical_table_map_id = "id" -> null

          - source {
              - physical_table_id = "id" -> null
            }
        }
      + logical_table_map {
          + alias                = "alias"
          + logical_table_map_id = "id"

          + source {
              + data_set_arn      = (known after apply)
              + physical_table_id = "id"
            }
        }

      - physical_table_map {
          - physical_table_map_id = "id" -> null

          - relational_table {
              - data_source_arn = "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:datasource/rds-datasource" -> null
              - name            = "table" -> null

              - input_columns {
                  - name = "name" -> null
                  - type = "STRING" -> null
                }
              - input_columns {
                  - name = "value" -> null
                  - type = "STRING" -> null
                }
            }
        }
      + physical_table_map {
          + physical_table_map_id = "id"

          + relational_table {
              + data_source_arn = "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:datasource/insights-rds-datasource"
              + name            = "table"
              + schema          = "schema"

              + input_columns {
                  + name = "name"
                  + type = "STRING"
                }
              + input_columns {
                  + name = "value"
                  + type = "STRING"
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

```

Sometimes the `logical_table_map` has changed, sometimes it hasn't.

Examination of the Terraform state file shows the following (truncated for brevity):

```json
{
  "version": 4,
  "terraform_version": "1.4.5",
  "serial": 53,
  "lineage": "cb7d28e5-44c1-410a-0941-7b2cea8844f8",
  "outputs": {},
  "resources": [
    {
      "mode": "managed",
      "type": "aws_quicksight_data_set",
      "name": "test",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "arn": "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:dataset/test",
            "aws_account_id": "ACCOUNT_ID",
            "column_groups": [],
            "column_level_permission_rules": [],
            "data_set_id": "test",
            "data_set_usage_configuration": [
              {
                "disable_use_as_direct_query_source": false,
                "disable_use_as_imported_source": false
              }
            ],
            "field_folders": [],
            "id": "ACCOUNT_ID,test",
            "import_mode": "DIRECT_QUERY",
            "logical_table_map": [
              {
                "alias": "alias",
                "data_transforms": [],
                "logical_table_map_id": "id",
                "source": [
                  {
                    "data_set_arn": "",
                    "join_instruction": [],
                    "physical_table_id": "id"
                  }
                ]
              }
            ],
            "name": "Test",
            "physical_table_map": [
              {
                "custom_sql": [],
                "physical_table_map_id": "id",
                "relational_table": [
                  {
                    "catalog": "",
                    "data_source_arn": "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:datasource/insights-rds-datasource",
                    "input_columns": [
                      {
                        "name": "name",
                        "type": "STRING"
                      },
                      {
                        "name": "value",
                        "type": "STRING"
                      }
                    ],
                    "name": "table",
                    "schema": ""
                  }
                ],
                "s3_source": []
              }
            ],
            "row_level_permission_data_set": [],
            "row_level_permission_tag_configuration": [],
            "tags": {},
            "tags_all": {}
          },
          "sensitive_attributes": [],
          "private": "bnVsbA=="
        }
      ]
    }
  }
}
```

2 things to note:

First, the `schema` field has not been populated with the expected value of "schema" which would seem to be the cause of constant changes for the `physical_table_map` block.

Second, the `logical_table_map.alias` field is correctly recorded as "alias" however the plan seems to indicate that the remote believes this value to be "table" instead. Invoking the AWS CLI confirms this:

```shell
$ aws quicksight describe-data-set --aws-account-id ACCOUNT_ID --data-set-id test
{
    "Status": 200,
    "DataSet": {
        "Arn": "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:dataset/test",
        "DataSetId": "test",
        "Name": "Test",
        "CreatedTime": "2023-04-13T16:30:42.274000+01:00",
        "LastUpdatedTime": "2023-04-13T16:38:03.544000+01:00",
        "PhysicalTableMap": {
            "id": {
                "RelationalTable": {
                    "DataSourceArn": "arn:aws:quicksight:eu-west-1:ACCOUNT_ID:datasource/rds-datasource",
                    "Name": "table",
                    "InputColumns": [
                        {
                            "Name": "name",
                            "Type": "STRING"
                        },
                        {
                            "Name": "value",
                            "Type": "STRING"
                        }
                    ]
                }
            }
        },
        "LogicalTableMap": {
            "id": {
                "Alias": "table",
                "Source": {
                    "PhysicalTableId": "id"
                }
            }
        },
        "OutputColumns": [
            {
                "Name": "name",
                "Type": "STRING"
            },
            {
                "Name": "value",
                "Type": "STRING"
            }
        ],
        "ImportMode": "DIRECT_QUERY",
        "ConsumedSpiceCapacityInBytes": 0,
        "DataSetUsageConfiguration": {
            "DisableUseAsDirectQuerySource": false,
            "DisableUseAsImportedSource": false
        }
    },
    "RequestId": "cab41098-5041-47cd-a5aa-7ecf45d4d557"
}
```

The expected behaviour here is not clear to me.

### Terraform Configuration Files

```terraform
provider "aws" {
  region = "eu-west-1"
}

resource "aws_quicksight_data_source" "db" {
  data_source_id = "rds-datasource"
  name           = "Database"
  type           = "POSTGRESQL"

  parameters {
    rds {
      database    = "public"
      instance_id = "test"
    }
  }

  credentials {
    credential_pair {
      username = "username"
      password = "password"
    }
  }
}

resource "aws_quicksight_data_set" "test" {

  data_set_id    = "test"
  import_mode    = "DIRECT_QUERY"
  name           = "Test"

  physical_table_map {
    physical_table_map_id = "id"
    relational_table {
      data_source_arn = aws_quicksight_data_source.db.arn
      name = "table"
      schema = "schema"
      input_columns {
        name = "name"
        type = "STRING"
      }
      input_columns {
        name = "value"
        type = "STRING"
      }
    }
  }

  logical_table_map {
    logical_table_map_id = "id"
    alias = "alias"
    source {
      physical_table_id = "id"
    }
  }
}
```

### Steps to Reproduce

1. Setup an RDS Postgres DB for use in step 2.
2. Plan & apply the Terraform configuration above.

### References

https://github.com/hashicorp/terraform-provider-aws/pull/30349

### Would you like to implement a fix?

None

**Comment by g-dx** [comment]
Related issues and PRs:


* https://github.com/hashicorp/terraform-provider-aws/issues/30702 concerns the missing `schema` handling.
* https://github.com/hashicorp/terraform-provider-aws/pull/30704 fixes that issue.

However, `alias` handling is likely still an issue.

**Comment by g-dx** [comment]
Can confirm that this is resolved in v4.63.0 of the provider.
